chat_shell/chat_shell/tools/knowledge_factory.py

```
@@ -207,3 +253,74 @@ async def get_knowledge_base_meta_list(
+                    set_span_attribute("rag_enabled", True)
+                    set_span_attribute("kb_id_with_rag", item.get("id"))
+                    return True
+
+            logger.info(
+                f"[knowledge_factory] No KB has RAG enabled among {knowledge_base_ids}"
+            )
+            add_span_event("no_rag_enabled")
+            set_span_attribute("rag_enabled", False)
+            return False
+
+    except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)